projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <UrtLib.h>
+#include <PwmLib.h>
+#include "Timer.h"
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <string.h>
+#include "DioLib.h"
+#include "math.h"
+#include "Communication.h"
+#include "CN0411.h"
+#include "AD7124_regs.h"
+
+/******************************************************************************/
+/*************************** Variable Definitions *****************************/
+/******************************************************************************/
+
+/* Available commands */
+char *cmd_commands[] = {
+	"help",
+	"syscal",
+	"refres",
+	"convmod",
+	"autoset",
+	"setdac",
+	"gainres",
+	"rtdval",
+	"pwmfreq",
+	"cellconst",
+	"solution",
+	"temp",
+	"vinput",
+	"readdac",
+	"rdr20s",
+	"rdr200s",
+	"rdres",
+	"cond",
+	"tds"
+};
+
+/* Functions for available commands */
+cmd_func cmd_fun[] = {
+	CN0411_cmd_help,
+	CN0411_cmd_sys_calib,
+	CN0411_cmd_off_res,
+	CN0411_cmd_conv_mode,
+	CN0411_cmd_autoset,
+	CN0411_cmd_set_dac,
+	CN0411_cmd_gain_res,
+	CN0411_cmd_rtd_val,
+	CN0411_cmd_pwm_freq,
+	CN0411_cmd_cell_const,
+	CN0411_cmd_solution,
+	CN0411_cmd_temp,
+	CN0411_cmd_vinput,
+	CN0411_cmd_read_dac,
+	CN0411_cmd_read_20s,
+	CN0411_cmd_read_200s,
+	CN0411_cmd_rdres,
+	CN0411_cmd_cond,
+	CN0411_cmd_tds,
+	NULL
+};
+
+/******************************************************************************/
+/************************* Function Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * CN0411 set DAC output value
+ *
+ * @param cn0411_dev - The device structure.
+ * @param output_val - voltage value to be written on DAC
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_DAC_set_value(cn0411_device *cn0411_dev, float output_val)
+{
+	int32_t ret;
+	uint32_t dac_val;
+
+	dac_val = (uint32_t)(output_val * DAC_FS_VAL / VREF);
+	cn0411_dev->ad5683_dev.dac_reg_value = dac_val;
+	ret = AD5683_write_dac_value(&cn0411_dev->ad5683_dev,
+				     cn0411_dev->ad5683_dev.dac_reg_value);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC Operation Mode function
+ *
+ * @param cn0411_dev - The device structure.
+ * @param mode - operation mode of the ADC
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_operation_mode (cn0411_device * cn0411_dev, opMode mode)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_ADC_Control].value =
+		(AD7124_ADC_CTRL_REG_REF_EN
+		 | AD7124_ADC_CTRL_REG_DATA_STATUS
+		 | AD7124_ADC_CTRL_REG_POWER_MODE(LOW_POWER)
+		 | AD7124_ADC_CTRL_REG_MODE(mode)
+		 | AD7124_ADC_CTRL_REG_CLK_SEL(INTERNAL_CLK1));
+
```

Comment:
I would rather use a mask here for better readability. Something like this:
`unsigned int mask;`
```
mask = (AD7124_ADC_CTRL_REG_REF_EN | AD7124_ADC_CTRL_REG_DATA_STATUS |
          AD7124_ADC_CTRL_REG_POWER_MODE(LOW_POWER) |
          AD7124_ADC_CTRL_REG_MODE(mode) |
          AD7124_ADC_CTRL_REG_CLK_SEL(INTERNAL_CLK1));

cn0411_dev->ad7124_dev.regs[AD7124_ADC_Control].value = mask;
```